pageserver/src/tenant.rs

```diff
@@ -1304,41 +1304,55 @@ impl Tenant {
     }
 
     async fn load_timeline_metadata(
+        self: Arc<Tenant>,
+        timeline_id: TimelineId,
+        remote_storage: GenericRemoteStorage,
+        cancel: CancellationToken,
+    ) -> anyhow::Result<TimelinePreload> {
+        let client = RemoteTimelineClient::new(
+            remote_storage.clone(),
+            self.deletion_queue_client.clone(),
+            self.conf,
+            self.tenant_shard_id,
+            timeline_id,
+            self.generation,
+        );
+        let timeline_preload = async move {
+            debug!("starting index part download");
+
+            let index_part = client.download_index_file(&cancel).await;
+
+            debug!("finished index part download");
+
+            Result::<_, anyhow::Error>::Ok(TimelinePreload {
+                client,
+                timeline_id,
+                index_part,
+            })
+        }
+        .map(move |res| {
+            res.with_context(|| format!("download index part for timeline {timeline_id}"))
+        })
+        .instrument(info_span!("download_index_part", %timeline_id))
+        .await?;
+
+        Ok(timeline_preload)
+    }
+
+    async fn load_timelines_metadata(
```

Comment:
I don't really see the point; also why this order:

1. only callsite
2. inner loop
3. outer loop

Reply:
reordered the functions in the source code.